packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -219,6 +257,72 @@ impl Resolver {
                 self.name_map[node.idx() + 1].insert(key, ref_);
             }
         }
+
+        self.add_implicit_index_resolutions(&FlatRootOrFragment::Fragment(flat_fragment));
+
+        // If the new nodes are index resolutions for a given parent,
+        // add them to `index_resolutions`
+        if let Some(as_index_resolutions) = as_index_resolutions {
+            self.replace_index_resolutions(
+                as_index_resolutions.parent_idx,
+                &flat_fragment.children,
+                as_index_resolutions.replace_mode,
+            );
+        }
+    }
+
+    /// Find any elements in `flat_root_or_fragment` that are marked `CHILDREN_ARE_IMPLICIT_INDEX_RESOLUTIONS`
+    /// and add index resolutions so that their children will resolve as their indices
+    fn add_implicit_index_resolutions(&mut self, flat_root_or_fragment: &FlatRootOrFragment) {
```

Comment:
I think this would be better as a more general function where you pass in `CHILDREN_ARE_IMPLICIT_INDEX_RESOLUTIONS` instead of reference a global directly.

Reply:
Keep the global declaration, just pass it into the function so the function is self-contained (and more testable)